cppcache/src/CacheFactory.cpp

```diff
@@ -100,23 +100,25 @@ Cache CacheFactory::create() const {
   const auto& memberListForVersionStamp =
       std::ref(*(cacheImpl->getMemberListForVersionStamp()));
 
-  serializationRegistry->addType2(
+  serializationRegistry->addDataSerializableFixedIdType(
       std::bind(TXCommitMessage::create, memberListForVersionStamp));
 
-  serializationRegistry->addType(
-      //TODO: This looks like the only thing to do here, but I'm not sure
-      static_cast<int64_t>(DSCode::PdxType),
-      std::bind(PdxType::CreateDeserializable, std::ref(*pdxTypeRegistry)));
+  serializationRegistry->setDataSerializablePrimitiveType(
+      // TODO: This looks like the only thing to do here, but I'm not sure
+      std::bind(PdxType::CreateDeserializable, std::ref(*pdxTypeRegistry)),
```

Comment:
This should be refactored out. It is not really a primitive type, it just pretends to be one. Give it the same treatment we did for `PdxTypeHandler` and `DataSerializableHandler`. In fact rename `PdxTypeHander` to `PdxSerializableHandler`, the current name is confusing with `PdxType`, which is the meta-data about PDX objects. Then implement this `PdxType` logic as `PdxTypeHandler`.

Reply:
Will address in separate ticket: https://issues.apache.org/jira/browse/GEODE-5743